pkgs/development/compilers/edk2/default.nix

```diff
@@ -45,6 +45,18 @@ edk2 = buildStdenv.mkDerivation {
     sha256 = "1ps244f7y43afxxw6z95xscy24f9mpp8g0mfn90rd4229f193ba2";
   };
 
+  patches = [
+    # Pull upstream fix for gcc-11 build.
+    (fetchpatch {
+      name = "gcc-11-vla.patch";
+      url = "https://github.com/google/brotli/commit/0a3944c8c99b8d10cc4325f721b7c273d2b41f7b.patch";
+      sha256 = "10c6ibnxh4f8lrh9i498nywgva32jxy2c1zzvr9mcqgblf9d19pj";
+      # Apply submodule patch to subdirectory: "a/" -> "BaseTools/Source/C/BrotliCompress/brotli/"
+      stripLen = 1;
+      extraPrefix = "BaseTools/Source/C/BrotliCompress/brotli/";
```

Comment:
I think it would be a much better idea to devendor brotli together with openssl and jansson if possible.

Reply:
I'm a bit worried about `-fshort-wchar` used all over the place to compile `brotli` and others: `edk2> gcc  -c ... -fshort-wchar -fno-strict-aliasing -fwrapv -fno-delete-null-pointer-checks -g -O2  brotli/c/enc/metablock.c -o brotli/c/enc/metablock.o`. `edk2` also overrides standard headers like `stddef.h` and `stdlib.h`: https://github.com/tianocore/edk2/tree/master/MdeModulePkg/Library/BrotliCustomDecompressLib

Not sure if it's a bug or feature. Probably a feature given that `edk2` is a toolchain to generate PE-COFFs with 16-bit wchar string ABI.

We could probably reuse `brotli` attribute's `.src` and `.patch` inputs (is it something that occasionally doe in `nixpkgs`? I'm curious to look at the example). https://github.com/tianocore/edk2/blob/master/MdeModulePkg/Library/BrotliCustomDecompressLib/BrotliCustomDecompressLib.inf#L42 would have to be kept in sync with `brotli` updates. Sounds slightly fragile.

Reply:
> Not sure if it's a bug or feature. Probably a feature given that edk2 is a toolchain to generate PE-COFFs with 16-bit wchar string ABI.

Oh, then it could be a bit tricky.

> We could probably reuse brotli attribute's .src and .patch inputs (is it something that occasionally doe in nixpkgs? I'm curious to look at the example).

Yeah, we can do that. It would probably make it easier to keep brotli close to the normal nixpkgs version.

Not sure what the best option is here, you decide.

Reply:
I'd prefer keeping it as is for such a small change :)